### PR TITLE
[WIP]Move performance test scripts to test-infra

### PR DESCRIPTION
**Pull request by Fredy-Z**
**What this PR does, why we need it**:
Move scripts for performance testing from `serving` to `test-infra`, so `eventing` can also use it.

/hold

### CODE REVIEW

**Comment by srinivashegde86** [comment]
we should probably merge this PR and then create a new PR that adds that functionality to the scripts. These scripts are in use and approved already. Anything new here, will need to be approved by Matt and Victor. So, we will need an agreed design before we can code it up. WDYT?

**Comment by Fredy-Z** [comment]
> we should probably merge this PR and then create a new PR that adds that functionality to the scripts. These scripts are in use and approved already. Anything new here, will need to be approved by Matt and Victor. So, we will need an agreed design before we can code it up. WDYT?

Matt agreed orally in our last standup meeting that we can create a `cluster.properties` file in each benchmark, and have postsubmit jobs to automatically create clusters for any new benchmarks. I've implemented that function in this script. 
But sure, I'll refine the design proposal I wrote earlier, and set up a meeting with Matt and Victor next week to go over all the design decisions we've made.
Let's hold this PR until then.

File: scripts/README.md [hunk 1]
```diff
@@ -224,6 +224,53 @@ kubectl get pods || fail_test
...
+```bash
+source vendor/knative.dev/test-infra/scripts/performance-tests.sh
+
+function update_knative() {
+  echo ">> Updating serving"
+  ko apply -f config/ -f config/v1beta1 || abort "failed to apply serving"
+}
+
+function update_benchmark() {
+  echo ">> Updating benchmark $1"
```
Comment (srinivashegde86) [inline scripts/README.md]: add a comment on what is $1

File: scripts/performance-tests.sh [hunk 2]
```diff
@@ -0,0 +1,266 @@
...
+  gcloud config set core/account ${USER_NAME}
+  echo "Using secret defined in ${PERF_TEST_GOOGLE_APPLICATION_CREDENTIALS}"
+  gcloud auth activate-service-account ${USER_NAME} --key-file=${PERF_TEST_GOOGLE_APPLICATION_CREDENTIALS}
+  echo "Using gcloud project ${PROJECT_NAME}"
+  gcloud config set core/project ${PROJECT_NAME}
+}
+
+# Creates a new cluster.
+# Parameters: $1 - cluster name
+#             $2 - cluster zone/region
```
Comment (srinivashegde86) [inline scripts/performance-tests.sh]:
```suggestion
#             $2 - cluster region
```

File: scripts/performance-tests.sh [hunk 3]
```diff
@@ -0,0 +1,266 @@
...
+  echo ">> Setup User"
+  echo "Using gcloud user ${USER_NAME}"
+  gcloud config set core/account ${USER_NAME}
+  echo "Using secret defined in ${PERF_TEST_GOOGLE_APPLICATION_CREDENTIALS}"
+  gcloud auth activate-service-account ${USER_NAME} --key-file=${PERF_TEST_GOOGLE_APPLICATION_CREDENTIALS}
+  echo "Using gcloud project ${PROJECT_NAME}"
+  gcloud config set core/project ${PROJECT_NAME}
+}
+
+# Creates a new cluster.
```
Comment (srinivashegde86) [inline scripts/performance-tests.sh]:
```suggestion
# Creates a new regional cluster.
```

File: scripts/performance-tests.sh [hunk 4]
```diff
@@ -0,0 +1,266 @@
...
+# Parameters: $1 - cluster name
+function get_benchmark_name() {
+  # get benchmark_name by removing the prefix from cluster name, e.g. get "load-test" from "serving-load-test"
+  echo ${1#$REPO_NAME"-"}
+}
+
+# Get cluster name from the benchmark name.
+# Parameters: $1 - benchmark name
+function get_cluster_name() {
+  # cluster_name is [repo_name]-[benchmark_name], e.g. serving-load-test
```
Comment (srinivashegde86) [inline scripts/performance-tests.sh]: why do we need to prepend repo?

**Comment by Fredy-Z** [comment]
Close this PR since we have decided to start migrating the performance test-infra to `Go`.

/close

**Pull request closed without merging**